### ISSUE TITLE
config about wan

### ISSUE THREAD

**Issue by jiajiaxiaoshenskx** [yifan123/flow_grpo#86]
I would like to ask for guidance on configuring the WAN training script.
If I want to train WAN on 8 GPUs (each with 40GB of memory), what parameters should I set in the WAN script to fully utilize the hardware?

Thank you in advance for your help!

**Comment by jingw193** [comment]
> > Thank you very much for your attention.
> > Overall, the hyperparameters are quite similar to SD3. The main difference between SD3 and Wan is that the clip range needs to be slightly larger, around 1e-3. Training becomes more stable when the group size is set above 8.
> 
> Hi yifan, is there any chance you can share the training loss curve of wan on OCR? Or how much performance improvement can we expect?

Hi ThisisBillhe,

The training reward curve of Wan-14B on the OCR task is shown below. Due to time constraints, we only trained for about 50 steps. Nevertheless, the reward demonstrates a clear upward trend, suggesting that further improvements can be expected with additional training steps.


<img width="2528" height="1328" alt="Image" src="https://github.com/user-attachments/assets/5d4625a1-bdae-4b43-956b-bb8336e03e26" />

**Comment by pbhfcycssjlmm** [comment]
> > > 非常感谢您的关注。
> > > 总体而言，超参数与 SD3 非常相似。SD3 与 Wan 的主要区别在于裁剪范围需要略大一些，大约为 1e-3。当组大小设置为 8 以上时，训练会更加稳定。
> > 
> > 
> > 你好，yifan，能分享一下wan在OCR上的训练损失曲线吗？或者说，我们能期待多少性能提升？
> 
> 你好，ThisisBillhe，
> 
> Wan-14B 在 OCR 任务上的训练奖励曲线如下所示。由于时间限制，我们仅训练了约 50 步。尽管如此，奖励仍呈现出明显的上升趋势，这表明随着训练步数的增加，可以期待进一步的提升。
> 
> <img alt="图像" width="2000" height="1328" src="https://private-user-images.githubusercontent.com/81085635/479418652-5d4625a1-bdae-4b43-956b-bb8336e03e26.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg2ODQ3MzgsIm5iZiI6MTc1ODY4NDQzOCwicGF0aCI6Ii84MTA4NTYzNS80Nzk0MTg2NTItNWQ0NjI1YTEtYmRhZS00YjQzLTk1NmItYmI4MzM2ZTAzZTI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI0VDAzMjcxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0ZmE4NzIxY2E2MzIzNGY4ZWUxNDFhZWE0MThiZTVmMjBlMzYyYjI0OTJiMmEyMzMyM2ZmNDU1NjdhMjM3YjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.z85sThlp6KncRYi6MWHREuNuzXM9jYcQ4Yx4o86DXPY">

您好，能否提供一下用FSDP改造后的训练脚本（Python文件）？我怀疑是我Wrap的block粒度不对，OOM错误发生在：
 File "/home/ning/flow_grpo/scripts/train_wan2_1_fsdp.py", line 711, in main
    neg_prompt_embed = compute_text_embeddings([""], text_encoders, tokenizers, max_sequence_length=512, device=device)
Tried to allocate 500.75 GiB (GPU 1; 60.97 GiB total capacity; 8.52 GiB already allocated; 8.52 GiB current active; 50.45 GiB free; 9.15 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.

My mapping function for wrapping：

`def get_transformer_layer_cls_for_wan():
    from diffusers.models.transformers.transformer_wan import WanTransformerBlock 
    from diffusers.models.attention_processor import Attention         
    from diffusers.models.attention import FeedForward             
    from transformers.models.umt5.modeling_umt5 import (
        UMT5Block,
        UMT5LayerSelfAttention,
        UMT5LayerFF,
    )

    return {
        WanTransformerBlock,
        Attention,
        FeedForward,
        UMT5Block, UMT5LayerSelfAttention, UMT5LayerFF
    }`

**Comment by pbhfcycssjlmm** [comment]
> > > Thank you very much for your attention.
> > > Overall, the hyperparameters are quite similar to SD3. The main difference between SD3 and Wan is that the clip range needs to be slightly larger, around 1e-3. Training becomes more stable when the group size is set above 8.
> > 
> > 
> > Hi yifan, is there any chance you can share the training loss curve of wan on OCR? Or how much performance improvement can we expect?
> 
> Hi ThisisBillhe,
> 
> The training reward curve of Wan-14B on the OCR task is shown below. Due to time constraints, we only trained for about 50 steps. Nevertheless, the reward demonstrates a clear upward trend, suggesting that further improvements can be expected with additional training steps.
> 
> <img alt="Image" width="2000" height="1328" src="https://private-user-images.githubusercontent.com/81085635/479418652-5d4625a1-bdae-4b43-956b-bb8336e03e26.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg2ODQ3MzgsIm5iZiI6MTc1ODY4NDQzOCwicGF0aCI6Ii84MTA4NTYzNS80Nzk0MTg2NTItNWQ0NjI1YTEtYmRhZS00YjQzLTk1NmItYmI4MzM2ZTAzZTI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI0VDAzMjcxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0ZmE4NzIxY2E2MzIzNGY4ZWUxNDFhZWE0MThiZTVmMjBlMzYyYjI0OTJiMmEyMzMyM2ZmNDU1NjdhMjM3YjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.z85sThlp6KncRYi6MWHREuNuzXM9jYcQ4Yx4o86DXPY">

Hello, could you provide the training script (Python file) after refactoring with FSDP? I suspect the granularity of my wrapped blocks is incorrect, and the OOM error occurs at：

 File "flow_grpo/scripts/train_wan2_1_fsdp.py", line 711, in main
    neg_prompt_embed = compute_text_embeddings([""], text_encoders, tokenizers, max_sequence_length=512, device=device)
Tried to allocate 500.75 GiB (GPU 1; 60.97 GiB total capacity; 8.52 GiB already allocated; 8.52 GiB current active; 50.45 GiB free; 9.15 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.

My mapping function for wrapping：

def get_transformer_layer_cls_for_wan():
    from diffusers.models.transformers.transformer_wan import WanTransformerBlock 
    from diffusers.models.attention_processor import Attention         
    from diffusers.models.attention import FeedForward             
    from transformers.models.umt5.modeling_umt5 import (
        UMT5Block,
        UMT5LayerSelfAttention,
        UMT5LayerFF,
    )

    return {
        WanTransformerBlock,
        Attention,
        FeedForward,
        UMT5Block, UMT5LayerSelfAttention, UMT5LayerFF
    }

**Comment by RobbinW** [comment]
> > > Thank you very much for your attention.
> > > Overall, the hyperparameters are quite similar to SD3. The main difference between SD3 and Wan is that the clip range needs to be slightly larger, around 1e-3. Training becomes more stable when the group size is set above 8.
> > 
> > 
> > Hi yifan, is there any chance you can share the training loss curve of wan on OCR? Or how much performance improvement can we expect?
> 
> Hi ThisisBillhe,
> 
> The training reward curve of Wan-14B on the OCR task is shown below. Due to time constraints, we only trained for about 50 steps. Nevertheless, the reward demonstrates a clear upward trend, suggesting that further improvements can be expected with additional training steps.
> 
> <img alt="Image" width="2000" height="1328" src="https://private-user-images.githubusercontent.com/81085635/479418652-5d4625a1-bdae-4b43-956b-bb8336e03e26.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA0ODQ2NjEsIm5iZiI6MTc3MDQ4NDM2MSwicGF0aCI6Ii84MTA4NTYzNS80Nzk0MTg2NTItNWQ0NjI1YTEtYmRhZS00YjQzLTk1NmItYmI4MzM2ZTAzZTI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA3VDE3MTI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMTkxYjkwMzg4NGNhZDVhMDNlMDIzZjFiMDNmZjlmZTBlZGU3NTlmMGQ3YjcxOTEzZDViZDM4Njk5Njg5YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hV1X4YYZZwThl296LuVyUAt-8uFvJnzmE9hdnhzJdRA">

hi jing, could you share your config for training wan-14B? I try to reproduce this  process on 8 GPU (80 G), but the reward barely increases

**Comment by zengkaiya** [comment]
> > > Thank you very much for your attention.
> > > Overall, the hyperparameters are quite similar to SD3. The main difference between SD3 and Wan is that the clip range needs to be slightly larger, around 1e-3. Training becomes more stable when the group size is set above 8.
> > 
> > 
> > Hi yifan, is there any chance you can share the training loss curve of wan on OCR? Or how much performance improvement can we expect?
> 
> Hi ThisisBillhe,
> 
> The training reward curve of Wan-14B on the OCR task is shown below. Due to time constraints, we only trained for about 50 steps. Nevertheless, the reward demonstrates a clear upward trend, suggesting that further improvements can be expected with additional training steps.
> 
> <img alt="Image" width="2000" height="1328" src="https://private-user-images.githubusercontent.com/81085635/479418652-5d4625a1-bdae-4b43-956b-bb8336e03e26.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzMxOTcyMDEsIm5iZiI6MTc3MzE5NjkwMSwicGF0aCI6Ii84MTA4NTYzNS80Nzk0MTg2NTItNWQ0NjI1YTEtYmRhZS00YjQzLTk1NmItYmI4MzM2ZTAzZTI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzExVDAyNDE0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhODM4ZmE5MzI4YTIzNmE3Yjg2Mzg2YTk3MWU5YzRkYzYyZTM2NjZiNDQ4YTYyNjFmZDNkNzhhODkwNWJkOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.k2CzFue3-bjNK4IGanisc_pWdBdqH6flCDCeSCc81NE">

您好，请教一下您在80G的卡上跑过14B模型的fsdp训练吗，我改造了下发现经常oom，可以借鉴下您的fsdp训练脚本吗？感谢🙏 @jingw193
